mypy/nodes.py

```diff
@@ -1604,16 +1811,19 @@ def calculate_mro(self) -> None:
 
         Raise MroError if cannot determine mro.
         """
-        self.mro = linearize_hierarchy(self)
+        mro = linearize_hierarchy(self)
+        if mro is None: return  # TODO: Or raise MroError()?
```

Comment:
Raising `MroError` seems better than just silently failing. We may have to process it somewhere, but if we have a serialized AST, the MRO should always be good.